### PR TITLE
Support for JUnit 5's DisplayName

### PR DESCRIPTION
**Pull request by BaerMitUmlaut**
This PR reads JUnit 5's `@DisplayName` annotation and shows it in the side bar instead of the method names:

screenshot

### CODE REVIEW

**Comment by jdneo** [comment]
@BaerMitUmlaut Thank you for the contribution.

We are doing refactoring these days, will review this PR later.

**Comment by BaerMitUmlaut** [comment]
@jdneo I took a quick glance over your refactors, I really like it so far, much cleaner it seems. Can you let me know when I should adapt my PR to them?

**Comment by jdneo** [comment]
@BaerMitUmlaut The refactor has almost done. We won't involve big changes in the near future. I think you can adapt this PR now. It will be great if you would like to resolve the conflicts. 👍 

Sorry that the refactoring causing you to have to reimplement it. And thank you for your contribution.

**Comment by BaerMitUmlaut** [comment]
Checkstyle seems to complain, but I can't figure out why.

I had to add some code specifically referencing JUnit5 in TestSearchUtils, which could be avoided by letting the searchers return the TestItems themselves, I think that would make more sense, too. However, that would be another rather big refactor.

**Comment by jdneo** [comment]
Sorry I was on vacation before, so the reply is not in time.

You can find the checkstyle errors in the output: 

```
[ERROR] /home/travis/build/Microsoft/vscode-java-test/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java:244:35: Variable 'annotation' should be declared final. [FinalLocalVariable]
[WARN] /home/travis/build/Microsoft/vscode-java-test/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java:280: 'if' construct must use '{}'s. [NeedBraces]
[WARN] /home/travis/build/Microsoft/vscode-java-test/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java:282: 'else' construct must use '{}'s. [NeedBraces]
```

File: com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java [hunk 1]
```diff
@@ -237,7 +239,15 @@ public static boolean isTestableClass(IType type) throws JavaModelException {
 
     public static TestItem constructTestItem(IJavaElement element, TestLevel level, TestKind kind)
             throws JavaModelException {
-        return new TestItem(element.getElementName(), parseTestItemFullName(element, level),
+        String displayName = element.getElementName();
+        if (kind == TestKind.JUnit5 && element instanceof IMethod) {
+            Optional<IAnnotation> annotation = getAnnotation((IMethod) element, DISPLAY_NAME_ANNOTATION_JUNIT5);
```
Comment (jdneo) [inline com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java]: add final according to the Checkstyle rule.

File: com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java [hunk 2]
```diff
@@ -251,31 +261,39 @@ public static TestKind resolveTestKindForMethod(IMethod method) {
...
-                return false;
+                return Optional.empty();
             }
 
             final String name = annotation.getElementName();
             final String[][] fullNameArr = method.getDeclaringType().resolveType(name);
             if (fullNameArr == null) {
                 final ICompilationUnit cu = method.getCompilationUnit();
-                return cu != null && cu.getImport(methodAnnotation).exists();
+                if (cu != null && cu.getImport(methodAnnotation).exists())
```
Comment (jdneo) [inline com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java]: add `{}`

File: com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java [hunk 3]
```diff
@@ -251,31 +261,39 @@ public static TestKind resolveTestKindForMethod(IMethod method) {
...
+                    return Optional.of(annotation);
+                else
```
Comment (jdneo) [inline com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestSearchUtils.java]: add `{}`

**Pull request merged**